src/MF_Modules/MFButton.cpp

```diff
@@ -17,31 +19,30 @@ void MFButton::update()
     uint8_t newState = (uint8_t) digitalRead(_pin);
     if (newState!=_state) {     
       _state = newState;
-      trigger();      
+      trigger(_state);      
```

Comment:
Why is a trigger method, which is just a wrapper around an if statement, which itself is just a wrapper around an event handler, which itself is just going to call cmdMessenger.sendCmd even required?

Why can't all those levels of indirection be removed and `update()` simply makes the calls to cmdMessenger directly?

Reply:
That is a question I would have gladly asked myself...!
I didn't want to push changes too far beyond what was really effective, because I didn't want to look too subversive :)
But I agree, in my opinion (see my previous reply to @elral) I would leave just the callback mechanism so as not to embed the details of cmdMessenger in this class. Unless @DocMoebiuz has anything against, it I'd be happy to go that way.

Reply:
Besides, event handlers for outgoing event messages mirror those required by cmdMessenger for incoming commands, which somehow makes sense from a logic standpoint at least.

Reply:
Ah, I see, that makes sense. Yes, so the individual MF things don't actually know about how the data is sent/received, and in the future some other transport mechanism could be used. Of course :)

Reply:
Yes, that makes sense.